File: 2244. Minimum Rounds to Complete All Tasks/main.cpp
#include <unordered_map>
#include <algorithm>
#include <numeric>

using namespace std;

class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        unordered_map<int, int> freq;
        for (int task : tasks) {
            freq[task]++;
        }

        int result = 0;
        for (auto& [task, count] : freq) {
            if (count == 1) {
                return -1;
            }
            result += (count + 2) / 3;
        }

        return result;
    }
};

File: 2244. Minimum Rounds to Complete All Tasks/main.ts
function minimumRounds(tasks: number[]): number {

        const freq = new Map(
            tasks.map(x => [x, tasks.filter(y => y === x).length])
        );

        let result = 0;
        for (const count of freq.values()) {
            if (count === 1) {
                return -1;
            }
            result += Math.floor((count + 2) / 3);
        }

        return result;
    
}